compiler/src/iree/compiler/Codegen/Common/TransformExtensions/CommonExtensions.cpp

```diff
@@ -1083,5 +1083,45 @@ void transform_dialect::GpuDistributeSharedMemoryCopyOp::getEffects(
   transform::modifiesPayload(effects);
 }
 
+//===----------------------------------------------------------------------===//
+// CreateTileSizesOp
+//===----------------------------------------------------------------------===//
+
+DiagnosedSilenceableFailure transform_dialect::CreateTileSizesOp::applyToOne(
+    transform::TransformRewriter &rewriter, Operation *target,
+    transform::ApplyToEachResultList &results,
+    transform::TransformState &state) {
+  Builder builder(getTarget().getContext());
+  ArrayRef<int64_t> shape =
+      llvm::cast<ShapedType>(target->getResult(0).getType()).getShape();
+  assert(shape.size() < 3);
+  SmallVector<int64_t> sz0, sz1;
+  SmallVector<Attribute> paramsArray0, paramsArray1;
+
+  // TODO: cover all shape cases
+  // (sz0[0] / sz1[0]) * (sz0[1] / sz1[1]) = 8
+  if (shape == ArrayRef<int64_t>({8192, 320})) {
+    sz0 = {512, 64}, sz1 = {64, 64};
+  } else if (shape[0] == 128) {
+    assert(shape[1] % 64 == 0);
+    sz0 = {128, 64}, sz1 = {32, 32};
+  } else {
+    // default tiling config
+    assert(shape[0] % 256 == 0 && shape[1] % 128);
```

Comment:
This, and the above asserts should be `return DiagnosedSilenceableFailure::definitFailure();`, or even better `emitDefiniteFailure() << "a nice error message for why it failed"`.